src/shared/model/MatchMiner.ts

```diff
@@ -0,0 +1,117 @@
+export interface ITrial {
+    id: string;
+    nctId: string | '';
+    protocolNo: string | '';
+    phase: string;
+    shortTitle: string;
+    status: string;
+    treatmentList: {
+        step: IStep[];
+    };
+}
+
+export interface IStep {
+    match?: IMatch[];
+    arm?: IArm[];
+}
+
+export interface IArm {
+    arm_description: string | ''; // Arm full name.
+    arm_type: string | ''; // Arm type(Control Arm)
+    arm_eligibility?: string;
+    arm_info?: string; // Real arm description.
+    drugs?: IDrug[];
+    match: IMatch[];
+}
+
+export interface IDrug {
+    name: string;
+    ncit_code?: string;
+    synonyms?: string;
+}
+
+export interface IMatch {
+    [key: string]: IClinical[] | IGenomic[] | IAnd[] | IOr[];
+}
+
+export interface IAnd {
+    and: IClinical[] | IGenomic[] | IAnd[] | IOr[];
+}
+
+export interface IOr {
+    or: IClinical[] | IGenomic[] | IAnd[] | IOr[];
+}
+export interface IClinical {
+    age_numerical: string,
+    oncotree_primary_diagnosis: string,
+    gender?: string,
+}
+
+export interface IGenomic {
+    hugo_symbol?: string,
+    annotated_variant?: string,
+}
+
+export interface ITrialMatch {
+    id: string;
+    nctId: string | '';
```

Comment:
do you need the `| ''` ?

Reply:
This field will be showed on the page. If it's null, I need to do checks which is not convenient so I set the default value to `''`.